src/any_agent/tracing/agent_trace.py

```diff
@@ -220,24 +220,12 @@ class AgentTrace(BaseModel):
     """A list of [`AgentSpan`][any_agent.tracing.agent_trace.AgentSpan] that form the trace.
     """
 
-    final_output: str | BaseModel | None = Field(default=None)
```

Comment:
I suppose this is inevitable but I guess it means you can't perfectly serialize and deserialize AgentTrace, since the pydantic model will get serialized as a dict and then when you de-serialize it, it will be loaded into final_output as a string, not as whatever type it was.

Reply:
we would need to use pickle if we want to preserve the actual object